lib/NeuraDialect/Transforms/Optimizations/HwAgnosticOpt/FoldConstantPass.cpp

```diff
@@ -233,6 +233,89 @@ struct FuseRemRhsConstantPattern : public FuseRhsConstantPattern<neura::RemOp> {
   }
 };
 
+// =========================================
+// FuseGepBaseConstantPattern
+// Fold constant base pointer for GEP operation
+// =========================================
+struct FuseGepBaseConstantPattern : public OpRewritePattern<neura::GEP> {
+  using OpRewritePattern<neura::GEP>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(neura::GEP gep_op,
+                                PatternRewriter &rewriter) const override {
+    Value base = gep_op.getBase();
+    
+    // Check if base exists and is a constant
+    if (!base || !isOriginConstantOp(base)) {
+      return failure();
+    }
+
+    auto constant_op = dyn_cast<neura::ConstantOp>(base.getDefiningOp());
+    Attribute base_const_value = getOriginConstantValue(base);
+
+    // Get all indices (everything after base)
+    SmallVector<Value> indices;
+    for (Value operand : gep_op.getIndicesAndPredicate()) {
+      indices.push_back(operand);
+    }
+
+    // Create new GEP with no base but with const_base_ptr attribute
+    auto fused_gep = rewriter.create<neura::GEP>(
+        gep_op.getLoc(), 
+        gep_op.getResult().getType(),
+        /*base=*/nullptr,
+        indices);
+    addConstantAttribute(fused_gep, "const_base_ptr", base_const_value);
```

Comment:
Also can you put `TODO: Gather all the attribute -- https://github.com/coredac/dataflow/issues/145` above this line to indicate we need to avoid using string.